src/account.ts

```diff
@@ -67,6 +67,10 @@ interface ReceiptLogWithFailure {
     failure: ServerError;
 }
 
+function parseJsonFromRawResponse (response: Uint8Array): any {
+    return JSON.parse(response.toString());
```

Comment:
nit: you cannot assume that `Uint8Array.toString()` does expected thing. Need to either use `Buffer.from` to be explicit

Reply:
The `Buffer.from` happens in `viewFunction`:

```js
async viewFunction(
    contractId: string,
    methodName: string,
    args: any,
    { parse = parseJsonFromRawResponse } = {}
): Promise<any> {
    args = args || {};
    this.validateArgs(args);
    const result = await this.connection.provider.query(`call/${contractId}/${methodName}`, base_encode(JSON.stringify(args)));
    if (result.logs) {
        this.printLogs(contractId, result.logs);
    }
    return result.result && result.result.length > 0 && parse(Buffer.from(result.result));
}
```

Would we want to do it again here, or pass raw `result.result` to `parse`?

Reply:
I understand that it happens upstream in this case. But declared input type doesn't require it happening. So you either want to be explicit about accepting only `Buffer` or do `Buffer.from` (best option as it works in runtime)